test/parallel/test-http-host-header-ipv6-fail.js

```diff
@@ -12,28 +12,31 @@
 const common = require('../common');
 const assert = require('assert');
 const http = require('http');
+const net = require('net');
 
-const hostname = '::1';
-
-function httpreq() {
-  const req = http.request({
-    host: hostname,
-    port: server.address().port,
-    path: '/',
-    method: 'GET'
-  });
-  req.end();
-}
-
-if (!common.hasIPv6) {
-  console.error('Skipping test, no IPv6 support');
-  return;
+function createLocalConnection(options) {
+  options.host = undefined;
+  options.port = this.port;
+  options.path = undefined;
+  return net.createConnection(options);
 }
 
-const server = http.createServer(common.mustCall(function(req, res) {
-  assert.ok(req.headers.host, `[${hostname}]`);
+http.createServer(common.mustCall(function(req, res) {
+  this.__connections = this.__connections || 2;
+  assert.strictEqual(req.headers.host, req.url.slice(1));
   res.end();
-  server.close(true);
-}));
-
-server.listen(0, hostname, () => httpreq());
+  if (--this.__connections === 0)
+    this.close();
+}, 2)).listen(0, function() {
+  const address = this.address();
+  http.get({
```

Comment:
Could you parameterize the literal `2`s, and add a comment here that these are the `N` connection attempts?